virtcontainers/kata_agent.go

```diff
@@ -999,31 +999,24 @@ func (k *kataAgent) removeIgnoredOCIMount(spec *specs.Spec, ignoredMounts []Moun
 	return nil
 }
 
-func (k *kataAgent) replaceOCIMountsForStorages(spec *specs.Spec, volumeStorages []*grpc.Storage) error {
+func (k *kataAgent) replaceOCIMountsForStorages(spec *specs.Spec, volumeStorages map[string]*grpc.Storage) error {
 	ociMounts := spec.Mounts
 	var index int
 	var m specs.Mount
 
-	for i, v := range volumeStorages {
-		for index, m = range ociMounts {
-			if m.Destination != v.MountPoint {
-				continue
-			}
-
-			// Create a temporary location to mount the Storage. Mounting to the correct location
-			// will be handled by the OCI mount structure.
-			filename := fmt.Sprintf("%s-%s", uuid.Generate().String(), filepath.Base(m.Destination))
-			path := filepath.Join(kataGuestSharedDir(), filename)
+	for index, m = range ociMounts {
+		if _, ok := volumeStorages[m.Destination]; !ok {
```

Comment:
The side effect is that you can only have one mount for each destination. While it is not a common case, docker does allow users to define multiple mounts for the same destination. So I'd prefer to keep the semantics.